src/Arcus.WebApi.Security/Authorization/Jwt/JwtTokenReader.cs

```diff
@@ -55,33 +55,44 @@ public JwtTokenReader(TokenValidationParameters tokenValidationParameters, strin
         }
 
         /// <summary>
-        ///     Validates if the token is considered valid
+        ///     Validates if the token is considered valid.
         /// </summary>
         /// <param name="token">JWT token</param>
         public async Task<bool> IsValidTokenAsync(string token)
         {
-            OpenIdConnectConfiguration config = await _configManager.GetConfigurationAsync();
-
-            var validationParameters = _tokenValidationParameters ?? new TokenValidationParameters
-            {
-                ValidateAudience = true,
-                ValidAudience = _applicationId,
-                ValidateIssuer = false,
-                IssuerSigningKeys = config.SigningKeys,
-                ValidateLifetime = true
-            };
-
-            SecurityToken jwtToken;
             try
             {
-                _handler.ValidateToken(token, validationParameters, out jwtToken);
+                TokenValidationParameters validationParameters = await DetermineTokenValidationParametersAsync();
+                _handler.ValidateToken(token, validationParameters, out SecurityToken jwtToken);
             }
-            catch (Exception)
+            catch (Exception exception)
```

Comment:
~~What might be the reason of an exception that is thrown here ?
Do you want to 'hide' all kind of exceptions and then just return 'the token is not valid', and hide the actual cause of the exception ?~~

~~I'd say that you should catch specific exceptions: the ones that you know that can occur, and act accordingly.  For other exceptions, I think I would just let them bubble up the stack.~~

edit: I've been looking at the [documentation ](JwtSecurityTokenHandler )of that method and now I can understand why you catch on `Exception` :)